_resources/port1.0/group/npm-1.0.tcl

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+#
+# This PortGroup is mainly used for npm ports (e.g. npm7). It helps setting up conflicts/dependencies/etc.
+#
+# Usage:
+#
+#     PortGroup            npm 1.0
+#     npm.setup            version
+#
+# It also makes publically accessible which node version each NPM/yarn depends on.
+# This can be accessed via [node_version NPM_VERSION]
+# e.g. npm7: [node_version 7] returns 16 for nodejs16
+#      yarn: [node_version yarn]
+
+
+# Set the minimum and maximum versions of npm
+# Required for livecheck and for setting conflicts
+# Also used by javascript portgroup
+options npm.maximum_version npm.minimum_version
+default npm.maximum_version 7
+default npm.minimum_version 3
+
+
+# This publically makes accessible which node version goes with which npm/yarn.
+# N.B. the npm.major of yarn is "yarn"
+proc node_version {npm.major} {
+    global os.major
+
+    # The platform is already darwin, so no need to double check
+    if {${npm.major} <= 5 || ${os.major} < 13} {
```

Comment:
> It makes no difference if you have already set platforms darwin since MacPorts does not check the value of platforms.

Ah ok, I didn't realise that. Thank you.